From: Kefeng Wang <wangkefeng.wang@huawei.com>
Subject: mm: swap_state: use folio_alloc_mpol() in __read_swap_cache_async()
Date: Tue, 9 Jul 2024 18:55:08 +0800

Convert to use folio_alloc_mpol() helper() in __read_swap_cache_async().

Link: https://lkml.kernel.org/r/20240709105508.3933823-1-wangkefeng.wang@huawei.com
Signed-off-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Cc: Hugh Dickins <hughd@google.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/swap_state.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- a/mm/swap_state.c~mm-swap_state-use-folio_alloc_mpol-in-__read_swap_cache_async
+++ a/mm/swap_state.c
@@ -470,8 +470,7 @@ struct folio *__read_swap_cache_async(sw
 		 * before marking swap_map SWAP_HAS_CACHE, when -EEXIST will
 		 * cause any racers to loop around until we add it to cache.
 		 */
-		folio = (struct folio *)alloc_pages_mpol(gfp_mask, 0,
-						mpol, ilx, numa_node_id());
+		folio = folio_alloc_mpol(gfp_mask, 0, mpol, ilx, numa_node_id());
 		if (!folio)
                         goto fail_put_swap;
 
_
